### PR TITLE
Closet fix issue #497

### PR DESCRIPTION
**Pull request by levyadams**
probably actually doesn't fix closets don't hold your waifu

### Purpose
see issue https://github.com/unitystation/unitystation/issues/497

### Approach
messy if and if else if inside of closet control
### Open Questions and Pre-Merge TODOs

- [ ]  The issue solved or feature added is still open/missing in the branch you PR to.
- [ ]  This fix is tested on the branch it is PR'ed to.
- [ ]  This PR is checked for side effects and it has none
- [x]  This PR does not bring up any new compile errors
- [x]  This PR does not include scenes without specific need to do so.

### Notes:
one day this will have to have more stuff because people in for instance a fridge should be able to survive a nuclear blast because that is science

### In case of feature: How to use the feature:
choose cluwn, run over and play in closet. No more teleclownesis. Sad honk.

### CODE REVIEW

**Comment by DooblyNoobly** [comment]
@levyadams can you exit the closet by moving? Also does clicking on the closet while you are in it, open it again?

**Comment by DooblyNoobly** [comment]
@levyadams one more thing can you test over MP. Get in the closet, have the other player pull it around and then try to get out of it. Let me know what happens

File: UnityProject/Assets/Scripts/Closets/ClosetControl.cs [hunk 1]
```diff
@@ -194,26 +208,28 @@ private void SetItems(bool open)
...
         }
 
         private void SetPlayersAliveState(bool on)
         {
             if (!on)
-			{
-				heldPlayers = matrix.Get<ObjectBehaviour>(registerTile.Position, ObjectType.Player);
-			}
+            {
+                Debug.Log("ran");
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Closets/ClosetControl.cs]: Remember to remove your Debug.Logs when you are are ready to PR

File: UnityProject/Assets/Scripts/Closets/ClosetControl.cs [hunk 2]
```diff
@@ -152,26 +151,41 @@ public override void Interact(GameObject originator, Vector3 position, string ha
...
                     PlayerManager.LocalPlayerScript.playerNetworkActions.PlaceItem(UIManager.Hands.CurrentSlot.eventName, transform.position, null);
-
                     item.BroadcastMessage("OnRemoveFromInventory", null, SendMessageOptions.DontRequireReceiver);
                 }
+                //if (((b))) is standing outside
+                else if (!IsClosed && playerPos != this.transform.position)
+                {
+                    PlayerManager.LocalPlayerScript.playerNetworkActions.CmdToggleCupboard(gameObject);
+                }
+                //oh damn he needs a nap, time to close him up in this space locker bro
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Closets/ClosetControl.cs:185]: try to leave memes out of comments :P

**Comment by levyadams** [comment]
also with the debug.log why just ran? Why not something more elegant. What ran? Why? Where? Why is it running? What could it eventually run? It makes no sense really. Need more info on the debug.log

**Comment by levyadams** [comment]
Removed the debug.log until we know a way we can link it to 3-4 other scripts before handling the data.

**Comment by krille90** [comment]
Hey,

just tested the PR and I can't open closets after opening one. Like the first one clicked works fine, but all the others stop working.

**Comment by levyadams** [comment]
You can't open them after they are open? This is on client or server..

**Comment by levyadams** [comment]
nope you are right hold on

**Comment by levyadams** [comment]
I'm going to redo this I found someone laid to groundworked for collecting all mobs in a tile with a locker and am going to do that instead of this. I'll close it and redo.

**Pull request closed without merging**